design/external-authorization-design.md

```diff
@@ -0,0 +1,453 @@
+# External Authorization Support
+
+Status: Reviewing
+
+## Abstract
+This document describes a design for performing request authorization for virtual hosts hosted by Contour.
+
+## Background
+
+- [#432][5] Provide support for integrating with an external auth service.
+- [#2459][3] External auth background and goals.
+
+## Goals
+- Allow operators to integrate existing authorization services with Contour.
+- Support arbitrary types of authentication and authorization.
```

Comment:
We should spell out the the specific types this proposal incorporates.

Reply:
This is about creating the mechanism that will let anyone add specific types.